REFERENCE: QGA-7

KristineZelenivska opened the pull request:
DESCRIPTION: made routes and controllers for registration and login, also added auth route to server.js

Commit:
REFERENCE: QGA-6
   DESCRIPTION: made some small changes: authenticate.js.
   added few pakages

Commit:
REFERENCE: QGA-6
   DESCRIPTION: Made authentication middleware. also created file
   for authController and auth Route

Commit:
REFERENCE: QGA-7
   DESCRIPTION: Made authController with Login and Register functions,
   also routes for login and register in /routes/auth and in server.

Commit:
REFERENCE: QGA-7
   DESCRIPTION: made controller and route for registration and login,
   also added auth route to server.js.

Commit:
REFERENCE: QGA-7

Gruseckis commented on .env.template:
```diff
@@ -1,11 +0,0 @@
-MONGODB_URI='mongodb://localhost:27017/quiz-game'
```
Please restore this file

KristineZelenivska replied:
I changed name from .env.template to just .env. Should i return word .template to this file?

Gruseckis commented on middlewares/authenticate.js:
```diff
@@ -0,0 +1,39 @@
+import jwt from 'jsonwebtoken'
```
This file should not be a part of this pull request

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+import jwt from 'jsonwebtoken';
+import AppError from '../errors/AppError';
+import * as UserModel from '../models/UserModel';
```
You import 8 functions and only use 3. Please only import methods that you will use.

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+import AppError from '../errors/AppError';
+import * as UserModel from '../models/UserModel';
+
+const logger = require('../utils/logger')('logController');
```
Please name this logger so it is more meaningful like authController?

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+
+const register = async (req, res, next) => {
+    logger.log('debug', 'register: %j', req.body);
+    const { username, email, hashedPassword, name, surname, dateOfBirth, level } = req.body;
```
Instead of deconstructing all values separately just get the body from req and use it.

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+            dateOfBirth: dateOfBirth,
+            level: level, //???
+        });
+        console.log(req.body.level)
```
Please remove console logs

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+            name: name,
+            surname: surname,
+            dateOfBirth: dateOfBirth,
+            level: level, //???
```
level should be defaulted in schema

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+        });
+        console.log(req.body.level)
+        logger.log('info', `Successfully registered: ${req.body.username}`);
+        res.status(200).send({ payload: { message: 'Successfully registered' } });
```
According to swagger you need to return user back

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,64 @@
+        if (isPasswordsEqual) {
+            const token = jwt.sign(
+                {
+                    data: { username: user.username },
```
please also add email to token

Gruseckis commented on server.js:
```diff
@@ -5,6 +5,9 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import bodyParser from 'body-parser';
 import index from './routes/index';
+import authRouter from './routes/auth';
+import authenticate from './middlewares/authenticate'
```
Method is defined but never used

Gruseckis commented on server.js:
```diff
@@ -25,6 +28,9 @@ app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
 
 // Routing
+//from Kristina, use authenticate method to add authentication to your route.
```
Remove comment

Gruseckis commented on server.js:
```diff
@@ -5,6 +5,9 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import index from './routes/index';
+import authRouter from './routes/auth';
```
rename auth.js to something more meaningfull like authRouter

Commit:
REFERECNCE: QGA-7
   DESCRIPTION: removed authenticate file from this branch,
   changed the imported functions, changed logger and auth route names,
   removed console.log and some comments, returned back the user and
   added email to token

Commit:
REFERENCE: QGA-7
   DESCRIPTION: returned old name to .env file, deleted middleware
    import in server.js

Commit:
REFERENCE: QGA-7
   DESCRIPTION: restored userModel file, to resolve conflict

Commit:
REFERENCE: QGA-7
   DESCRIPTION: deleted "level" field in registration,
   so it could be default if needed

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,60 @@
+            dateOfBirth: body.dateOfBirth
+        });
+        logger.log('info', `Successfully registered: ${body.username}`);
+        res.status(200).send({ payload: { user: user } });
```
You can just do `{ payload: { user } }`

Commit:
REFERENCE: QGA-7
  DESCRIPTION: added level value

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,60 @@
+    try {
+        const user = await getUserByUsername(req.body.username);
+        if (!user) {
+            logger.log('debug', 'Login failed');
```
change level do error and message to something more informative like at what username login failed so we could get some statistics

Gruseckis commented on controllers/authController.js:
```diff
@@ -0,0 +1,60 @@
+            logger.log('info', `Successfully loged in: ${user.username}`);
+            res.status(200).send({ payload: { message: 'Successfully loged in', token } });
+        } else {
+            throw new AppError('CHECK TOKEN PART', 400);
```
Error message should be 'Wrong cridentials';

Commit:
REFERENCE: QGA-7
  DESCRIPTION: changed returned value of registration,
  added more info to logger error message
   and also changed error message for login
